### DISCUSSION TITLE
Custom domain on Github Pages

### DISCUSSION THREAD

**Discussion by jmgirard** [quarto-dev/quarto-cli#3249] [Q&A] [answered]
I am hosting a Quarto website on Github Pages using the "docs" method. I have a custom domain and thus need to have a `docs/CNAME` file in my project. However, it seems that quarto is removing this file whenever I render via `quarto render`. Is there a way to tell quarto to keep this CNAME file? I tried the `website: site-url` option but that didn't seem to do it.

**Comment by cscheid** [comment, marked as answer]
You can't in general control the contents of the output directory. But you should be able to move the CNAME file to the root of the quarto project, then add it as an explicit resource in your `_quarto.yml` configuration:

```yml
resources:
  - CNAME
```

**Reply by jmgirard** [reply]
That's a clever solution and seemed to work. It might be worth adding a note to https://quarto.org/docs/publishing/github-pages.html about this as the intended way to support custom domain names.

**Reply by blantonsbourbon** [reply]
I have tried but still not working.

**Reply by cscheid** [reply]
@Lov3Camille Can you share a full example where this doesn't work for you?

**Reply by blantonsbourbon** [reply]
sorry I wrongly input docs/CNAME, it works now.

**Comment by msh855** [comment]
I do not understand what I need to add. I know that the CNAME relates to the custom domain. But to add to the `docs\CNAME`, what is the type of CNAME? is a  `.txt`? `.qmd` ? and should be just an empty file or with the custom url in it ?

**Reply by jmgirard** [reply]
> I do not understand what I need to add. I know that the CNAME relates to the custom domain. But to add to the `docs\CNAME`, what is the type of CNAME? is a `.txt`? `.qmd` ? and should be just an empty file or with the custom url in it ?

It is a plain text file with no extension. It should contain your desired alias. For example, mine is: `affcom.ku.edu`
Read more about CNAMEs here: https://support.google.com/a/answer/47283?hl=en
